stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/gallery/AttachmentGalleryActivity.kt

```diff
@@ -250,13 +252,15 @@ public class AttachmentGalleryActivity : AppCompatActivity() {
 
     public companion object {
         private const val EXTRA_KEY_INITIAL_INDEX: String = "extra_key_initial_index"
+        private const val EXTRA_KEY_ATTACHMENT_URL_SIGNER: String = "extra_key_url_signer"
 
         internal const val EXTRA_ATTACHMENT_OPTION_RESULT = "extra_attachment_option_result"
 
         @JvmStatic
-        public fun createIntent(context: Context, initialIndex: Int): Intent {
+        public fun createIntent(context: Context, initialIndex: Int, attachmentUrlSigner: UrlSigner): Intent {
             return Intent(context, AttachmentGalleryActivity::class.java).apply {
                 putExtra(EXTRA_KEY_INITIAL_INDEX, initialIndex)
+                putExtra(EXTRA_KEY_ATTACHMENT_URL_SIGNER, attachmentUrlSigner)
```

Comment:
Do not like this approach, because it's easy forget to implement `Serializable` interface for custom `UrlSigner`. What if pass signer to view model constructor and sign image previews there?